src/rtc_pl031.rs

```diff
@@ -147,52 +161,77 @@ fn get_current_time() -> u32 {
     epoch_time.as_secs() as u32
 }
 
-impl Rtc<NoEvents> {
-    /// Creates a new `AMBA PL031 RTC` instance without any metric
-    /// capabilities.
-    ///
-    /// # Example
-    ///
-    /// You can see an example of how to use this function in the
-    /// [`Example` section from `Rtc`](struct.Rtc.html#example).
-    pub fn new() -> Rtc<NoEvents> {
-        Self::with_events(NoEvents)
-    }
-}
-
 impl Default for Rtc<NoEvents> {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<EV: RtcEvents> Rtc<EV> {
-    /// Creates a new `AMBA PL031 RTC` instance and invokes the `rtc_events`
-    /// implementation of `RtcEvents` during operation.
-    ///
-    /// # Arguments
-    /// * `rtc_events` - The `RtcEvents` implementation used to track the occurrence
-    ///                  of failure or missed events in the RTC operation.
-    pub fn with_events(rtc_events: EV) -> Self {
-        Rtc {
+// This is the state from which a fresh Rtc can be created.
+impl Default for RtcState {
+    fn default() -> Self {
+        RtcState {
             // The load register is initialized to 0.
             lr: 0,
-
             offset: 0,
-
             // The match register is initialised to zero (not currently used).
             // TODO: Implement the match register functionality.
             mr: 0,
-
             // The interrupt mask is initialised as not set.
             imsc: 0,
-
             // The raw interrupt is initialised as not asserted.
             ris: 0,
+        }
+    }
+}
 
+impl Rtc<NoEvents> {
+    /// Creates a new `AMBA PL031 RTC` instance without any metric
+    /// capabilities.
+    pub fn new() -> Self {
+        Self::from_state(&RtcState::default(), NoEvents)
+    }
+}
+
+impl<EV: RtcEvents> Rtc<EV> {
+    /// Creates a new `AMBA PL031 RTC` instance from a given `state` and invokes the `rtc_events`
+    /// implementation of `RtcEvents` during operation.
```

Comment:
I think we should also add a comment about wanting to start from a clean state, in which case people should use `with_events`.